spartan/aztec-network/templates/transaction-bot.yaml

```diff
@@ -123,6 +123,17 @@ spec:
               echo "AZTEC_NODE_URL=${AZTEC_NODE_URL}"
 
               export BOT_ACCOUNT_SALT=$(echo $K8S_POD_NAME | awk -F'-' '{print $NF+1}') # +1 so that salt is not 0
+
+              # a crude way to avoid a race condition that could arise if more than one bot is started:
+              # the bots need to register contract classes, if they all start at the same time (or close to it) then multiple of them might try to register the same class leading to failed txs due to duplicate nullifiers
+              # instead lets allow the first bot to start immediately giving it the responsibility to register the classes
+              # every other bot after that will sleep at least a slot before starting up
+              if [[ "$BOT_ACCOUNT_SALT" != "1" ]]; then
+                sleep_duration=$(({{ .Values.aztec.slotDuration }} + ($RANDOM % 10)))
```

Comment:
Nit: RANDOM may be zero, and the node may need a few seconds (after the block is mined) to sync and realise that the class was registered. Maybe we want to add 2-3 extra seconds in there just for good measure?